File: code/graph_algorithms/src/data_structures/adjacency_matrix.py
class graph:
    adj=[]
    def __init__(self,v,e):
        self.v=v
        self.e=e
        graph.adj=[[0 for i in range(v)] for i in range(v)]
        
    def adj_matrix(self,start,e):
        graph.adj[start][e]=1
        graph.adj[e][start]=1
        
    def display_adj_matrix(self):
        print("The adjacency matrix is: \n")
        for i in range (v):
            for j in range (v):
                print(graph.adj[i][j], end =" ")
            print("\n")
choice="y"
v=int(input("Enter number of vertices: "))
e=v
g = graph(v,e)
print("The vertices are: ")
for i in range (v):
    print(i, end=' ')
print('\n')
while choice == 'y':
    x = int(input("Enter vertex 1: "))
    y = int(input("Enter vertex 2: "))
    g.adj_matrix(x,y)
    choice = input("Do you want to add more edges? (y/n): ")
    if choice == "n":
        break
g.display_adj_matrix()
"""
Output:
Enter number of vertices: 3
The vertices are: 
0 1 2 

Enter vertex 1: 1
Enter vertex 2: 2
Do you want to add more edges? (y/n): y
Enter vertex 1: 0
Enter vertex 2: 2
Do you want to add more edges? (y/n): n
The adjacency matrix is: 

0 0 1 

0 0 1 

1 1 0 
"""

File: code/graph_algorithms/src/data_structures/adjacency_matrix.cpp
/*
 * Part of Cosmos by OpenGenus Foundation
 */
#include <iostream>
#include <vector>
using namespace std;

//Used for printing out the adjList after you finish building it
void printMatrix(vector<vector<int>> adjMatrix)
{
	cout << "The adjacency matrix is: " << endl;
	for (int i = 0; i < adjMatrix.size(); i++)
	{
		for (int j = 0; j < adjMatrix[i].size(); j++)
		{
			cout << adjMatrix[i][j] << " ";
		}
		cout << endl;
	}
}

int main()
{
	int nodeNum;
	int numInput;
	char charInput;
	cout << "Input the number of nodes in the graph" << endl;
	cin >> nodeNum;
	vector<vector<int>> adjMatrix(nodeNum, vector<int>(nodeNum)); //Initalizes a 2D vector of all 0s
	for (int i = 0; i < nodeNum; i++)
	{
		bool moveOn = false;
		while (!moveOn)
		{
			cout << "Enter an edge for vertex " << i + 1 << ": " << endl;
			cin >> numInput;
			if (numInput > nodeNum)
				cout << "Invalid input" << endl;
			else
				adjMatrix[i][numInput - 1] = 1;
			cout << "Do you want to add more edges? (y/n): " << endl;
			cin >> charInput;
			if (charInput == 'n')
				moveOn = true;
		}
	}
	//When printing the matrix, the rows is the source node and the columns are the destination
	printMatrix(adjMatrix);
	return 0;
}